components/openindiana/bhyve-fw/Makefile

```diff
@@ -0,0 +1,69 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2020 OmniOS Community Edition (OmniOSce) Association.
+#
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		uefi-edk2
+COMPONENT_VERSION=	20180309
+COMPONENT_FMRI=		system/bhyve/firmware
+COMPONENT_CLASSIFICATION=System/Core
+COMPONENT_SUMMARY=	UEFI-EDK2(+CSM) firmware for bhyve
+COMPONENT_SRC=		uefi-edk2-il-udk2014.sp1-1
+COMPONENT_ARCHIVE=	il-udk2014.sp1-1.tar.gz
+COMPONENT_ARCHIVE_HASH= \
+	sha256:d894d1ca61beeb0a1b76c2d6bbd137b02feb4783e4307f36e76efa29d415f6a8
+COMPONENT_ARCHIVE_URL= \
+	https://github.com/illumos/uefi-edk2/archive/$(COMPONENT_ARCHIVE)
+COMPONENT_PROJECT_URL=	https://github.com/illumos/uefi-edk2
+COMPONENT_LICENSE=	simplified-BSD
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/ips.mk
+
+COMPONENT_BUILD_ENV =
+COMPONENT_BUILD_ENV += GCCVER=4.4.4
+COMPONENT_BUILD_ENV += GCCPATH=/opt/gcc/4.4.4
+COMPONENT_BUILD_ENV += GMAKE=/usr/bin/gmake
+COMPONENT_BUILD_ENV += GAS=/usr/bin/gas
+COMPONENT_BUILD_ENV += GAR=/usr/bin/gar
+COMPONENT_BUILD_ENV += GLD=/usr/bin/gld
+COMPONENT_BUILD_ENV += GOBJCOPY=/usr/bin/gobjcopy
+
+$(BUILD_DIR_64)/.built: $(SOURCE_DIR)/.unpacked
+	$(MKDIR) -p $(BUILD_DIR_64)
```

Comment:
Should $(BUILD_DIR_64)/.built target just depend on individual  build targets ? Like in https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/components/editor/emacs/Makefile

Reply:
I can restructure it that way if you prefer - are you thinking of a DEBUG and RELEASE target?
You can probably tell that I'm not familiar with the build framework - I had to add the .unpacked dependency to get it to actually download and unpack the archive.
I'm happy to restructure it in whatever way you think is best but I'll need a couple of extra pointers.

Reply:
Yes. Usually you have `$(BUILD_DIR)/%/.built:  $(SOURCE_DIR)/.prep` dependency from one of the specific makefiles (justmake.mk, configure.mk, etc). Perhaps, you can use justmake.mk, overwrite COMPONENT_INSTALL_ACTION, COMPONENT_BUILD_ACTION (perhaps COMPONENT_COPY_ACTION  if you need) and perhaps defining two variants - like in https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/components/editor/vim/Makefile#L52. 
This also could be ugly, but perhaps avoids using so much shell code.
Or just split current build action in two - debug and release build.

Reply:
I'd looked at it myself, but for now I just have no time.

Reply:
No problem, thanks for the feedback, I'll take a look.